### PR TITLE
Activities/history

### PR DESCRIPTION
**Pull request by Neathan**

### CODE REVIEW

**Comment by Neathan** [comment]
History view should now be ready for review.

tsc currently gives an error that `dateStyle` is not part of `DateTimeFormatOptions`, which it should be.
Extract from lib.es2020.intl.d.ts:
```
interface DateTimeFormatOptions {
        dateStyle?: "full" | "long" | "medium" | "short";
        timeStyle?: "full" | "long" | "medium" | "short";
        calendar?: string;
        dayPeriod?: "narrow" | "short" | "long";
        numberingSystem?: string;
        hourCycle?: "h11" | "h12" | "h23" | "h24";
        fractionalSecondDigits?: 0 | 1 | 2 | 3;
    }
```
Anyone know whats happening here?

File: src/storage/activities.tsx [hunk 1]
```diff
@@ -87,10 +87,58 @@ export const useActivities = (): [Activities, ModifyActivities] => {
...
+   * Requires date to exist.
+   *
+   * @param date - The date on which the entry will be inserted (YYYY-mm-dd)
+   * @param score - The score of the day will be inserted [0-10]
+   * @returns `true` if the entry was inserted, `false` otherwise
+   */
+  const setRating = (date: string, score: number): boolean => {
+    if(isDate(date) && 0 <= score && score <= 10) {
+      const newActivities = [ ...activities ];
+      const index = newActivities.findIndex(a => a.date === date);
```
Comment (EMattfolk) [inline src/storage/activities.tsx:130]: This is not checked. It might be -1.

File: src/storage/activities.tsx [hunk 2]
```diff
@@ -87,10 +87,58 @@ export const useActivities = (): [Activities, ModifyActivities] => {
...
+   * This function will add the date if not present.
+   *
+   * @param date - The date on which the entry will be inserted (YYYY-mm-dd)
+   * @param startHour - The start hour on which the entries will be inserted [0, 22]
+   * @param endHour - The end hour on which the entries will be inserted [1, 23]
+   * @param entry - The entry to be inserted
+   * @returns `true` if the entry was inserted, `false` otherwise
+   */
+  const addInterval = (date: string, startHour: number, endHour: number, entry: ActivitiesEntry): boolean => {
+    if(isDate(date) && 0 <= startHour && startHour < endHour && endHour <= 24) {
```
Comment (EMattfolk) [inline src/storage/activities.tsx]:
```suggestion
    if (isDate(date) && 0 <= startHour && startHour < endHour && endHour <= 24) {
```

File: src/storage/activities.tsx [hunk 3]
```diff
@@ -87,10 +87,58 @@ export const useActivities = (): [Activities, ModifyActivities] => {
...
+      const newActivities = _insertDay(date);
+      const index = newActivities.findIndex(a => a.date === date);
+      for(let hour = startHour; hour < endHour; ++hour) {
```
Comment (EMattfolk) [inline src/storage/activities.tsx]:
```suggestion
      for (let hour = startHour; hour < endHour; ++hour) {
```

File: src/storage/activities.tsx [hunk 4]
```diff
@@ -87,10 +87,58 @@ export const useActivities = (): [Activities, ModifyActivities] => {
...
+   *
+   * @remarks
+   * Requires date to exist.
+   *
+   * @param date - The date on which the entry will be inserted (YYYY-mm-dd)
+   * @param score - The score of the day will be inserted [0-10]
+   * @returns `true` if the entry was inserted, `false` otherwise
+   */
+  const setRating = (date: string, score: number): boolean => {
+    if(isDate(date) && 0 <= score && score <= 10) {
```
Comment (EMattfolk) [inline src/storage/activities.tsx]:
```suggestion
    if (isDate(date) && 0 <= score && score <= 10) {
```

File: src/storage/activities.tsx [hunk 5]
```diff
@@ -87,10 +87,58 @@ export const useActivities = (): [Activities, ModifyActivities] => {
...
+        newActivities[index].entries[hour] = entry;
+      }
+      setStoreItem(activitiesKey, newActivities);
+      return true;
+    }
+    return false;
+  };
+
+  /**
+   * Modifies the score of a give date.
```
Comment (EMattfolk) [inline src/storage/activities.tsx]:
```suggestion
   * Modifies the score of a given date.
```

File: src/components/activity/ActivityHistory.tsx [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+  return date;
+};
+
+export const ActivityHistory = ({navigation}: any) => {
+  const [activities, modifyActivities] = Storage.useActivities();
+  const [settings, modifySettings] = Storage.useSettings();
+  const lang = useTranslation();
+  const { colors } = useTheme();
+
+  let historyItems: any = [];
```
Comment (EMattfolk) [inline src/components/activity/ActivityHistory.tsx]: Does this need to be any?

**Review by EMattfolk — APPROVED** [review]
It looks great in the app! The code also looks pretty good.

I would recommend modifying `_insertDay` so that it inserts things in sorted order. There was also no real way of changing the day.

**Comment by EMattfolk** [comment]
Should we close this one?

**Comment by malin-mallan** [comment]
> Should we close this one?

Ye but it need one more aproval

**Comment by Neathan** [comment]
Closing this issue since it has become redundant by #31

**Pull request closed without merging**